twistr/events.go

```diff
@@ -1141,112 +1150,227 @@ func PlayIranianHostageCrisis(s *State, player Aff) {
+		return SelectNInfluenceCheck(s, player,
+			"Remove 2 US influence from the Middle East", 2,
+			InRegion(MiddleEast), HasInfluence(USA))
+	})
+	RemoveInfluence(s, USA, cs)
 }
 
 func PlaySovietsShootDownKAL007(s *State, player Aff) {
 	/* Degrade the DEFCON level by 1 and the US receives 2 VP. The US may place
 	   influence or make Realignment rolls, using this card, if South Korea is US
 	   controlled. */
+	s.DegradeDefcon(1)
+	s.GainVP(USA, 2)
+	if s.Countries[SKorea].Controlled() == USA {
+		ConductOps(s, player, Cards[SovietsShootDownKAL007], INFLUENCE, REALIGN)
+	}
 }
 
 func PlayGlasnost(s *State, player Aff) {
 	/* Improve the DEFCON level by 1 and the USSR receives 2 VP. The USSR may
 	   make Realignment rolls or add Influence, using this card, if the “#87 – The
 	   Reformer” Event has already been played. */
+	s.ImproveDefcon(1)
+	s.GainVP(SOV, 2)
+	if s.Effect(TheReformer) {
+		ConductOps(s, player, Cards[Glasnost], REALIGN, INFLUENCE)
+	}
 }
 
 func PlayOrtegaElectedInNicaragua(s *State, player Aff) {
 	/* Remove all US Influence from Nicaragua. The USSR may make a free Coup
 	   Attempt, using this card’s Operations value, in a country adjacent to
 	   Nicaragua. */
+	nicaragua := s.Countries[Nicaragua]
+	nicaragua.Inf[USA] = 0
+	adjToNicaragua := make([]CountryId, len(nicaragua.AdjCountries))
+	for i, c := range nicaragua.AdjCountries {
+		adjToNicaragua[i] = c.Id
+	}
+	DoFreeCoup(s, player, Cards[OrtegaElectedInNicaragua], adjToNicaragua)
 }
 
 func PlayTerrorism(s *State, player Aff) {
 	/* The player’s opponent must randomly discard 1 card from their hand. If
 	   the “#82 – Iranian Hostage Crisis” Event has already been played, a US
 	   player (if applicable) must randomly discard 2 cards from their hand. */
+	opp := player.Opp()
+	if len(s.Hands[opp].Cards) == 0 {
+		return
+	}
+	card := SelectRandomCard(s, opp)
+	// XXX messaging
+	s.Hands[opp].Remove(card)
+	if opp == USA && s.Effect(IranianHostageCrisis) {
+		if len(s.Hands[opp].Cards) == 0 {
+			return
+		}
+		card := SelectRandomCard(s, opp)
+		s.Hands[opp].Remove(card)
+	}
 }
 
 func PlayIranContraScandal(s *State, player Aff) {
 	/* All US Realignment rolls, for the remainder of this turn, receive -1 to
 	   their die roll. */
+	s.TurnEvents[IranContraScandal] = player
 }
 
 func PlayChernobyl(s *State, player Aff) {
 	/* The US must designate a single Region (Europe, Asia, etc.) that, for the
 	   remainder of the turn, the USSR cannot add Influence to using Operations
 	   points. */
+	region := SelectRegion(s, player, "Choose a region where USSR is blocked from influencing for the turn")
+	s.TurnEvents[Chernobyl] = player
+	s.ChernobylRegion = region
 }
 
 func PlayLatinAmericanDebtCrisis(s *State, player Aff) {
 	/* The US must immediately discard a card with an Operations value of 3 or
 	   more or the USSR may double the amount of USSR Influence in 2 countries in
 	   South America. */
+	if hasInHand(s, USA, ExceedsOps(2)) &&
```

Comment:
I dunno man, I know we just edited this, but this is still feels like an ordeal to read.